### PR TITLE
lib/libc/math: Enhancing log & logf implementations

### PR DESCRIPTION
**Pull request by NilabhShant**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [fd07f045e2861e128968a6dd21aacd84669ec0f1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fd07f045e2861e128968a6dd21aacd84669ec0f1] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_math.c:2386: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)

File: lib/libc/math/lib_logf.c [hunk 1]
```diff
@@ -81,17 +83,25 @@ float logf(float x)
...
+        }
+
 	while (y > y_old + epsilon || y < y_old - epsilon) {
 		y_old = y;
 		ey = expf(y);
 		y -= (ey - x) / ey;
 
-		if (y > 700.0) {
-			y = 700.0;
+		if (y > FLT_MAX_EXP_X) {
```
Comment (jeongchanKim) [inline lib/libc/math/lib_logf.c]: The original r-value is 700.0, but you changed to 88.0F.
Can you tell me why did you change like this?

File: lib/libc/math/lib_logf.c [hunk 2]
```diff
@@ -81,17 +83,25 @@ float logf(float x)
 	iter = 0;
 	relax_factor = 1;
 
+	if (x < 0.0) {
+                return NAN;
```
Comment (JeongJunSik) [inline lib/libc/math/lib_logf.c]: Use tabs instead of spaces

File: lib/libc/math/lib_log.c [hunk 3]
```diff
@@ -84,6 +86,10 @@ double log(double x)
 		return NAN;
 	}
 
+	if (x == 0.0) {
+                return -HUGE_VAL;
```
Comment (JeongJunSik) [inline lib/libc/math/lib_log.c]: Use tabs instead of spaces

File: lib/libc/math/lib_logf.c (hunk 1 above)
Comment (NilabhShant) [reply lib/libc/math/lib_logf.c]: Max value of y in logf() is 88.722839 which will be round off to 88.0F same as NuttX logf().

File: lib/libc/math/lib_logf.c (hunk 2 above)
Comment (NilabhShant) [reply lib/libc/math/lib_logf.c]: Ok, I will do it.

File: lib/libc/math/lib_log.c (hunk 3 above)
Comment (NilabhShant) [reply lib/libc/math/lib_log.c]: Ok, I will do it.

File: lib/libc/math/lib_logf.c [hunk 4]
```diff
@@ -81,17 +83,25 @@ float logf(float x)
...
+        }
+
+	if (x == 0.0) {
+                return -HUGE_VAL;
```
Comment (yashwanth686007) [inline lib/libc/math/lib_logf.c]: I think its better to represent HUGE_VALF to indicate float value.

Comment (NilabhShant) [reply lib/libc/math/lib_logf.c]: It will be same.
But yes,I will do it.

**Comment by seinfra** [comment]
Target : [7f27d83f40b7420c75dc1db629e644522f3892b4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7f27d83f40b7420c75dc1db629e644522f3892b4] - Code Rule Check (C++) OK.
